README.md
# sbt-dependencies (ABANDONED)

## DEPRECATION NOTICE

**sbt-dependencies** is an SBT plugin that allows to you to keep your project dependencies up-to-date.

This plugin depends on the [`sbt-updates`](https://github.com/rtimush/sbt-updates) plugin

47 Degrees has discontinued the development and maintenance of this library. The source code is left here for those insterested in studying it.

[comment]: # (Start Badges)

[![Build Status](https://travis-ci.org/47degrees/sbt-dependencies.svg?branch=master)](https://travis-ci.org/47degrees/sbt-dependencies) [![Maven Central](https://img.shields.io/badge/maven%20central-0.4.0-green.svg)](https://repo1.maven.org/maven2/com/47deg/sbt-dependencies_2.12_1.0) [![License](https://img.shields.io/badge/license-Apache%202-blue.svg)](https://raw.githubusercontent.com/47degrees/sbt-dependencies/master/LICENSE) [![Join the chat at https://gitter.im/47deg/sbt-dependencies](https://badges.gitter.im/47deg/sbt-dependencies.svg)](https://gitter.im/47deg/sbt-dependencies?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) [![GitHub Issues](https://img.shields.io/github/issues/47degrees/sbt-dependencies.svg)](https://github.com/47degrees/sbt-dependencies/issues)

[comment]: # (End Badges)

*Latest version*

[comment]: # (Start Replace)
```scala
addSbtPlugin("com.47deg" % "sbt-dependencies" % "0.4.0")
```

[comment]: # (End Replace)

## sbt-dependencies in the wild

If you wish to add your library here please consider a PR to include it.

[comment]: # (Start Copyright)
# Copyright

sbt-dependencies is designed and developed by 47 Degrees

Copyright (C) 2018 47 Degrees. <http://47deg.com>

[comment]: # (End Copyright)


CHANGELOG.md
# Changelog

## 12/14/2018 - Version 0.4.0

Release changes:

* Removes support for SBT 0.13.x ([#51](https://github.com/47degrees/sbt-dependencies/pull/51))


## 11/16/2017 - Version 0.3.9

Release changes:

* Bumps sbt-org-policies ([#50](https://github.com/47degrees/sbt-dependencies/pull/50))


## 11/16/2017 - Version 0.3.8

Release changes:

* Fixes sbt-org-policies-core version ([#47](https://github.com/47degrees/sbt-dependencies/pull/47))
* Removes scala 2.10 from cross scala versions ([#49](https://github.com/47degrees/sbt-dependencies/pull/49))


## 11/16/2017 - Version 0.3.7

Release changes:

* Fixes sbt-org-policies-core version ([#46](https://github.com/47degrees/sbt-dependencies/pull/46))


## 11/15/2017 - Version 0.3.6

Release changes:

* Bumps sbt-org-policies, sbt-updates, and releases 0.3.6 ([#44](https://github.com/47degrees/sbt-dependencies/pull/44))


## 10/23/2017 - Version 0.3.5

Release changes:

* Bumps sbt-org-policies version ([#40](https://github.com/47degrees/sbt-dependencies/pull/40))
* Adds the org policies default settings to the projectSettings ([#39](https://github.com/47degrees/sbt-dependencies/pull/39))
* Bumps sbt-org-policies plugin ([#41](https://github.com/47degrees/sbt-dependencies/pull/41))
* Set crossScalaVersion to 2.12 and force org-policies-core version ([#42](https://github.com/47degrees/sbt-dependencies/pull/42))
* Fixes GitHub token ([#43](https://github.com/47degrees/sbt-dependencies/pull/43))


## 10/02/2017 - Version 0.3.4

Release changes:

* Enables microsite plugin again ([#38](https://github.com/47degrees/sbt-dependencies/pull/38))


## 10/02/2017 - Version 0.3.3

Release changes:

* Makes site publication through Github4s ([#37](https://github.com/47degrees/sbt-dependencies/pull/37))


## 10/02/2017 - Version 0.3.2

Release changes:

* Upgrade github4s sbt org policies ([#36](https://github.com/47degrees/sbt-dependencies/pull/36))


## 09/19/2017 - Version 0.3.1

Release changes:

* Release 0.3.1 ([#35](https://github.com/47degrees/sbt-dependencies/pull/35))


## 09/18/2017 - Version 0.3.0

Release changes:

* Release 0.3.0 ([#34](https://github.com/47degrees/sbt-dependencies/pull/34))


## 09/13/2017 - Version 0.2.0

Release changes:

* Port plugin to sbt {1.0.x, 0.13.16} ([#30](https://github.com/47degrees/sbt-dependencies/pull/30))
* Release 0.2.0 ([#31](https://github.com/47degrees/sbt-dependencies/pull/31))
* Missing space in if condition ([#32](https://github.com/47degrees/sbt-dependencies/pull/32))
* Remove depUpdateDependencyIssues in travis.yml ([#33](https://github.com/47degrees/sbt-dependencies/pull/33))


## 04/25/2017 - Version 0.1.2

Release changes:

* Fixes publish microsites command ([#28](https://github.com/47degrees/sbt-dependencies/pull/28))
* Upgrades sbt-org-policies ([#29](https://github.com/47degrees/sbt-dependencies/pull/29))


## 04/20/2017 - Version 0.1.1

Release changes:

* Modifies the file committing ([#25](https://github.com/47degrees/sbt-dependencies/pull/25))
* Removes pull command ([#26](https://github.com/47degrees/sbt-dependencies/pull/26))
* Fixes git remote command ([#24](https://github.com/47degrees/sbt-dependencies/pull/24))
* Adds a basic microsite ([#27](https://github.com/47degrees/sbt-dependencies/pull/27))

project/ProjectPlugin.scala
import com.typesafe.sbt.site.SitePlugin.autoImport._
import microsites.MicrositesPlugin.autoImport._
import sbt.Keys._
import sbt._
import sbt.ScriptedPlugin.autoImport._
import sbtorgpolicies.OrgPoliciesPlugin
import sbtorgpolicies.OrgPoliciesPlugin.autoImport._
import sbtorgpolicies.model._
import sbtorgpolicies.templates.badges._

object ProjectPlugin extends AutoPlugin {

  override def trigger: PluginTrigger = allRequirements

  override def requires: Plugins = OrgPoliciesPlugin

  object autoImport {

    lazy val pluginSettings = Seq(
      sbtPlugin := true,
      scalaVersion := scalac.`2.12`,
      crossScalaVersions := Seq(scalac.`2.12`),
      libraryDependencies ++= Seq(%%("github4s", "0.19.0"), %%("org-policies-core", "0.9.4")),
      addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.5.0"),
      scriptedLaunchOpts := {
        scriptedLaunchOpts.value ++
          Seq(
            "-Xmx2048M",
            "-XX:ReservedCodeCacheSize=256m",
            "-XX:+UseConcMarkSweepGC",
            "-Dplugin.version=" + version.value,
            "-Dscala.version=" + scalaVersion.value
          )
      }
    )

    lazy val micrositeSettings = Seq(
      micrositeName := "sbt-dependencies",
      micrositeDescription := "An SBT plugin that allows to you to keep your project dependencies up-to-date",
      micrositeBaseUrl := "sbt-dependencies",
      micrositeCompilingDocsTool := WithTut,
      micrositeDocumentationUrl := "/sbt-dependencies/docs/",
      micrositeGithubOwner := "47deg",
      micrositeGithubRepo := "sbt-dependencies",
      micrositeHighlightTheme := "darcula",
      micrositePalette := Map(
        "brand-primary"   -> "#005b96",
        "brand-secondary" -> "#03396c",
        "brand-tertiary"  -> "#011f4b",
        "gray-dark"       -> "#48474C",
        "gray"            -> "#8D8C92",
        "gray-light"      -> "#E3E2E3",
        "gray-lighter"    -> "#F4F3F9",
        "white-color"     -> "#FFFFFF"
      ),
      includeFilter in makeSite := "*.html" | "*.css" | "*.png" | "*.jpg" | "*.gif" | "*.js" | "*.swf" | "*.md"
    )

  }

  override def projectSettings: Seq[Def.Setting[_]] =
    Seq(
      name := "sbt-dependencies",
      orgGithubSetting := GitHubSettings(
        organization = "47degrees",
        project = (name in LocalRootProject).value,
        organizationName = "47 Degrees",
        groupId = "com.47deg",
        organizationHomePage = url("http://47deg.com"),
        organizationEmail = "hello@47deg.com"
      ),
      scalaVersion := "2.12.10",
      startYear := Option(2017),
      orgMaintainersSetting := List(Dev("fedefernandez", Some("Fede Fernández"))),
      orgBadgeListSetting := List(
        TravisBadge.apply(_),
        MavenCentralBadge.apply(_),
        LicenseBadge.apply(_),
        GitHubIssuesBadge.apply(_)
      )
    )
}
